### PR TITLE
Update common files and fix lint warnings

### PR DESCRIPTION
**Pull request by rlenglet**
Update common files (`make updatecommon`).

Remove duplicate `lint` and `format` targets from Makefile in favor of the ones from `Makefile.common.mk`. Remove duplicate `docker.*` targets in `tools/istio-cni-docker.mk`.

Define and run `lint` (`make lint`) and `shellcheck` CircleCI jobs.

Fix `make lint` warnings.
Temporarily disable the new `shellcheck` job, which is currently failing. Fixing it is done in #111.

### CODE REVIEW

**Comment by geeknoid** [comment]
This looks good.

How about adding a CircleCI job to check the lint status? Just steal the config from another repo's CircleCI config.

**Comment by rlenglet** [comment]
@geeknoid 
> How about adding a CircleCI job to check the lint status? Just steal the config from another repo's CircleCI config.

The CircleCI `build` job already ran `make lint`. But there was no job running `make test`. I'll separate `make lint` into a separate `lint` job and add a `test` job, for consistency.

**Comment by geeknoid** [comment]
That'll be great.

And you may wish to update the common files again, since I've just pushed some minor updates.

**Comment by rlenglet** [comment]
I'll temporarily disable the new `test` and `shellcheck`, which are currently failing. Fixing them will require more work, so I'll do that in separate PRs.

**Comment by tiswanso** [comment]
FYI, the existing install-cni job was running `make test`

**Comment by tiswanso** [comment]
Looks like this PR is syncing istio/cni with the evolved istio/istio best practices, right?

File: .circleci/config.yml [hunk 1]
```diff
@@ -175,7 +175,31 @@ jobs:
...
+
+  lint:
+    <<: *defaults
+    steps:
+      - checkout
+      - run:
+          name: Run linters
+          command: make lint
+
+  test:
```
Comment (tiswanso) [inline .circleci/config.yml]: IMO, don't add a new test job as the existing install-cni job is for performing `make test`.

Comment (rlenglet) [reply .circleci/config.yml]: Done.

**Comment by geeknoid** [comment]
Could you delete bin/fmt.sh and bin/linters.sh as part of this change, since they will no longer be used?

Thanks.

**Comment by rlenglet** [comment]
> Could you delete bin/fmt.sh and bin/linters.sh as part of this change, since they will no longer be used?

Good catch, I missed those. Done.

**Comment by rlenglet** [comment]
@tiswanso, thanks for the approval. Any idea why Mergify still fails and prevents from merging this?

**Comment by john-a-joyce** [comment]
@rlenglet - it looks like it is complaining about needing reviews form everyone you requested.  Perehaps you an reduce the list of requested reviewers and it should help

**Comment by tiswanso** [comment]
@rlenglet -- trying to figure that out as well.  The error details look like the job last ran 4 days ago so I tried retriggering manually--it seems it is queued.  Here's the "current" error (4 days old):

```
Branch protection settings are blocking automatic merging

GitHub error message: Waiting on code owner review from baodongli, costinm, john-a-joyce, jwendell, linsun, rcurran1, rshriram, tiswanso, and/or venilnoronha.
```

**Comment by rlenglet** [comment]
@john-a-joyce I can't edit the list of reviewers for this PR. It's set automatically from the contents of `CODEOWNERS`.

**Comment by john-a-joyce** [comment]
that's a new behavior previously it wouldn't indicate all teh code owners needed to review

**Comment by tiswanso** [comment]
@john-a-joyce  -- the mergify tool should just require 1 of them via this config:  https://github.com/istio/cni/blob/master/.mergify.yml#L6

**Comment by rlenglet** [comment]
> that's a new behavior previously it wouldn't indicate all teh code owners needed to review

It only needs one of the reviewers to approve. As @tiswanso mentioned, the problem seems to be that the Mergify job hasn't run for 4 days. Let me try to make a trivial change to a commit to see if it will trigger this time.

**Comment by john-a-joyce** [comment]
I was just reading the error message but I guess the "and/or" at the end doesn't imply everyone has to review it.  Sorry about the noise

**Comment by rlenglet** [comment]
Eh, it got triggered this time. It's unfortunate that one has to push a change after geting approval, just to get Mergify to work. :/

**Pull request merged**